microcontroller/src/board.c

```diff
@@ -0,0 +1,175 @@
+#include "board.h"
+#include "stm32f7xx_hal.h"
+
+// Board Initialization: Set all pins to input floating (Except SWD)
+void Board_Init(void) {
+  // Define the GPIO configuration struct
+  // This struct defines Pin, Mode, Pull (Up/Down), Speed & Alternate Function
+  GPIO_InitTypeDef GPIO_InitStruct;
+
+/*********************************** GPIO A ***********************************/
+
+  // GPIOA Setup : All pins will be initialized to floating inputs (except SWD)
+  // GPIOA RCC Clock should be enabled
+  __HAL_RCC_GPIOA_CLK_ENABLE();
+
+  // PA0-PA12 & PA15: Floating Input
+  GPIO_InitStruct.Mode = GPIO_MODE_INPUT;
+
+  // PA0-PA12 & PA15: No Pull Up / Pull Down
+  GPIO_InitStruct.Pull = GPIO_NOPULL;
+
+  // PA0-PA12 & PA15: Highest Speed
+  GPIO_InitStruct.Speed = GPIO_SPEED_FREQ_VERY_HIGH;
+
+  // PA0-PA12 & PA15: We can either apply OR of the 13 pins;
+  // or we can use: PIN_All XOR (PIN_14 | PIN_13)
+  GPIO_InitStruct.Pin = GPIO_PIN_All ^ (GPIO_PIN_13 | GPIO_PIN_14);
+
+  // Apply the Configuration to GPIOA
+  HAL_GPIO_Init(GPIOA, &GPIO_InitStruct);
+
+  // PA13: Alternate Function Push Pull
+  GPIO_InitStruct.Mode = GPIO_MODE_AF_PP;
+
+  // PA13: Pull Up (SWDIO)
+  GPIO_InitStruct.Pull = GPIO_PULLUP;
+
+  // PA13: Highest Speed
+  GPIO_InitStruct.Speed = GPIO_SPEED_FREQ_VERY_HIGH;
+
+  // PA13: Alternate Function 0 (SWD & JTAG)
+  GPIO_InitStruct.Alternate = GPIO_AF0_SWJ;
+
+  // PA13: Since the convention in this file is to use pin numbers,
```

Comment:
Why did u take up this convention?
In general, it seems more intuitive to me to use the actual function names of the pins